fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/sfcc/flaps_channel.rs

```diff
@@ -252,7 +277,119 @@ impl FlapsChannel {
             );
     }
 
+    fn powerup_reset(&mut self, adirs: &impl AdirsMeasurementOutputs) {
+        // Auto Command restart
+        if self.csu_monitor.get_last_valid_detent() != CSU::Conf1 {
+            self.flap_auto_command_active = false;
+        } else {
+            // The match can be shortened by a convoluted if statement however
+            // I believe it would make debugging and understanding the state machine harder
+            match self.get_cas(adirs) {
```

Comment:
I didn't realize in the initial review of the SFCC but there are some cases in this `match` block which will never execute.
For example
```rust
(Some(cas1), _) if cas1 > self.kts_100
    && cas1 < self.kts_210
    && SlatFlapControlComputerMisc::between_0_1f_enlarged_target_range(
        self.flaps_feedback_angle,
    )
```
and
```rust
(_, Some(cas2)) if cas2 > self.kts_100
    && cas2 < self.kts_210
    && SlatFlapControlComputerMisc::between_0_1f_enlarged_target_range(
        self.flaps_feedback_angle,
    )
```
have the exact same condition but the latter will never execute because either both (`cas1` and `cas2`) have some value or both have `None` (because you're calling `self.get_cas`).